vm/src/vm.rs

```diff
@@ -624,14 +624,26 @@ impl VirtualMachine {
 
     // get_method should be used for internal access to magic methods (by-passing
     // the full getattribute look-up.
-    pub fn get_method(&self, obj: PyObjectRef, method_name: &str) -> PyResult {
+    pub fn get_method_or_type_error(
+        &self,
+        obj: PyObjectRef,
+        method_name: &str,
+        err_msg: String,
```

Comment:
Could you make `err_msg` here a `FnOnce() -> String` instead of just a `String`? That way we can avoid unnecessary allocations when there's no error.